### PR TITLE
tests: fatal: convert legacy test to ztest

### PR DESCRIPTION
**Pull request by punitvara**
Clear checkpatch errors and make use of ztest apis to
support ztest framework.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Comment by punitvara** [comment]
@SavinayDharmappa Can you please review ?

File: tests/kernel/fatal/src/main.c [hunk 1]
```diff
@@ -36,7 +37,7 @@ static volatile int crash_reason;
 #ifndef CONFIG_ARM
 FUNC_NORETURN
 #endif
-void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF *pEsf)
+void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF * pEsf)
```
Comment (andrewboie) [inline tests/kernel/fatal/src/main.c]: why was a space inserted here, that's not correct style

Comment (punitvara) [reply tests/kernel/fatal/src/main.c]: Checkpatch throws following error if I don't do this:

[pv@pvara-mobl1 new_pr]$ ./scripts/checkpatch.pl -f tests/kernel/fatal/src/main.c 
tests/kernel/fatal/src/main.c:40: ERROR:SPACING: need consistent spacing around '*' (ctx:WxV)

Comment (nashif) [reply tests/kernel/fatal/src/main.c]: leave it as is, there is another way to fix this. (see #4714)

Comment (punitvara) [reply tests/kernel/fatal/src/main.c]: @nashif Done

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**